tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs

```diff
@@ -85,6 +85,43 @@ public async Task SuccessfulResponseFromLocalImdsAsync()
             ValidateInstanceMetadata(regionalMetadata);
         }
 
+        [TestMethod]
+        public void MultiThreadSuccessfulResponseFromLocalImds_HasOnlyOneImdsCall()
```

Comment:
I removed the semaphore from the RegionManager and this test still passed. So it can't reliably test the related code... Although I'm not sure of a better way to test this, since it involves a specific order and time of threads executing the same code. I did test it manually by debugging through 2 threads, and everything worked as expected though.

Reply:
If we find a better way to test it, I will be more than happy to improve it. But sounds like good for now.